sjb/config/test_cases/ami_build_origin_int_rhel_fork.yml

```diff
@@ -0,0 +1,223 @@
+        oct prepare docker --version=1.12.6 --repo=oso-rhui-rhel-server-* -u https://mirror.openshift.com/enterprise/rhel/dockerextra/x86_64/os/
+
+      else
+        echo "[FATAL] No Docker version is known for \`${TARGET_ORIGIN_BRANCH}\`.
+              Contact Steve Kuznetsov <skuznets@redhat.com> for assistance."
+        exit 1
+      fi
+  - type: "host_script"
+    title: "prepare repositories"
+    script: |-
+      oct prepare repositories
+  - type: "script"
+    title: "install system accounting rules"
+    script: |-
+      cat <<CONF >origin-ami-accounting.conf
+      [Manager]
+      DefaultCPUAccounting=yes
+      DefaultMemoryAccounting=yes
+      CONF
+      sudo mkdir -p /etc/systemd/system.conf.d/
+      sudo mv origin-ami-accounting.conf /etc/systemd/system.conf.d/
+      sudo systemctl daemon-reexec
+  - type: "host_script"
+    title: "package the AMI"
+    script: |-
+      oct package ami --upgrade
+  - type: "script"
+    title: "log system information"
+    script: |-
+      go version
+      docker version
+  - type: "script"
+    title: "use a ramdisk for etcd"
+    script: |-
+      sudo mkdir -p /tmp
+      sudo mount -t tmpfs -o size=2048m tmpfs /tmp
+      sudo restorecon -R /tmp
+  - type: "script"
+    title: "build an origin release"
+    repository: "origin"
+    script: |-
+      export OS_BUILD_IMAGE_ARGS=''
+      hack/build-base-images.sh
+      OS_ONLY_BUILD_PLATFORMS='linux/amd64' OS_BUILD_ENV_PRESERVE="_output/local/releases:_output/local/bin" hack/env make build-rpms
+      hack/build-images.sh
+      hack/extract-release.sh
+  - type: "script"
+    title: "verify commit history"
+    repository: "origin"
+    script: |-
+      # run commitchecker outside release container as it needs
+      # access to git; also explicitly force godeps verification
+      branch="$( git rev-parse --abbrev-ref --symbolic-full-name HEAD )"
+      if [[ "${branch}" == "master" ]]; then
+        RESTORE_AND_VERIFY_GODEPS=1 make verify-commits -j
+      fi
+  - type: "script"
+    title: "run check and verify tasks"
+    repository: "origin"
+    script: |-
+      OS_BUILD_ENV_EXCLUDE='_output/local/go' hack/env TEST_KUBE='true' JUNIT_REPORT='true' make check -o build -j -k
+  - type: "script"
+    title: "run integration tests"
+    repository: "origin"
+    script: |-
+      OS_BUILD_ENV_PRESERVE=_output/local/bin/linux/amd64/integration.test hack/env make build-integration-test
+      OPENSHIFT_SKIP_BUILD='true' JUNIT_REPORT='true' make test -o check -o build -k
+  - type: "host_script"
+    title: "make a Trello comment if requested"
+    script: |-
+        set +o nounset +o errexit
+        if [[ -n "$TRELLO_URL" ]]; then
+          trello comment "A fork ami has been created for this card: \`${TAG_NAME}_${BUILD_NUMBER}\`" --card-url $TRELLO_URL
+        fi
+  - type: "host_script"
+    title: "release the AMI"
+    script: |-
+      oct package ami --mark-ready
```

Comment:
We need to make sure that these fork AMIs are not used by the rest of the process -- `oct provision` will look for tags on the AMI to determine operating system, image stage, readiness, etc -- we need to probably create a new image stage `fork` for these? Not sure what the best approach is, but I did not realize this would be an issue until just now -- we'll need to change code in `origin-ci-tool` to support this. One of the most simple and otherwise useful approaches may be to allow `oct package ami` to take a variable number of `--tag KEY=VALUE` type flags that we stick on the AMI as-is, and amend the `oct provision` logic to ensure that there isn't a Trello URL in the tags or something, instead of trying to add in a new stage.